components/scream/src/physics/share/physics_test_data.hpp

```diff
@@ -72,25 +74,69 @@ struct SHOCGridData : public PhysicsTestData {
 #define PTD_ASSIGN_OP(name, num_scalars, ...)                                  \
   name& operator=(const name& rhs) { PTD_ASS##num_scalars(__VA_ARGS__); assignment_impl(rhs); return *this; }
 
+#define PTD_DIM_RENAME1(ndim1)                                              Int ndim1() const { return dim1; }
+#define PTD_DIM_RENAME2(ndim1, ndim2)        PTD_DIM_RENAME1(ndim1);        Int ndim2() const { return dim2; }
+#define PTD_DIM_RENAME3(ndim1, ndim2, ndim3) PTD_DIM_RENAME2(ndim1, ndim2); Int ndim3() const { return dim3; }
+
+#define PTD_DIM_RENAME(num_args, ...)           \
+  PTD_DIM_RENAME##num_args(__VA_ARGS__)
+
+#define PTD_ADD_1_0 1
+#define PTD_ADD_1_1 2
+#define PTD_ADD_1_2 3
+#define PTD_ADD_1_3 4
+#define PTD_ADD_1_4 5
+#define PTD_ADD_1_5 6
+#define PTD_ADD_1_6 7
+#define PTD_ADD_1_7 8
+#define PTD_ADD_1_8 9
+#define PTD_ADD_1_9 10
+
+#define PTD_ADD_2_0 2
+#define PTD_ADD_2_1 3
+#define PTD_ADD_2_2 4
+#define PTD_ADD_2_3 5
+#define PTD_ADD_2_4 6
+#define PTD_ADD_2_5 7
+#define PTD_ADD_2_6 8
+#define PTD_ADD_2_7 9
+#define PTD_ADD_2_8 10
+#define PTD_ADD_2_9 11
+
+#define PTD_ADD_3_0 3
+#define PTD_ADD_3_1 4
+#define PTD_ADD_3_2 5
+#define PTD_ADD_3_3 6
+#define PTD_ADD_3_4 7
+#define PTD_ADD_3_5 8
+#define PTD_ADD_3_6 9
+#define PTD_ADD_3_7 10
+#define PTD_ADD_3_8 11
+#define PTD_ADD_3_9 12
+
+#define PTD_STD_DEF(name, dim, num_scalars, ...)        \
+  PTD_DATA_COPY_CTOR(name, PTD_ADD_##dim##_##num_scalars);  \
```

Comment:
I know I'm late to the party, but out of curiosity: why did you need the ADD_x_y macro? Didn't something like
```
PTD_DATA_COPY_CTOR(name, (dim + num_scalars) );
```
work?

Reply:
Nvm, I think I understand. The sum of the two needs to be used to append to other macro names, so it must be a simple numeral, not an expression. Got it.